### ISSUE TITLE
Data Augmentation for YOLOv8

### ISSUE THREAD

**Issue by Zengyf-CVer** [ultralytics/ultralytics#3401]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

@glenn-jocher I found a file about data augmentation, how to customize this file in Python code? Which parameter needs to be set?
![image](https://github.com/ultralytics/ultralytics/assets/41098760/d088a188-eea3-4c7e-987b-5ef36a5bf50f)
![image](https://github.com/ultralytics/ultralytics/assets/41098760/5cfb7806-5905-47d7-b8bb-397b46654e12)

**Comment by glenn-jocher** [comment]
@Zengyf-CVer for augmentation hyps see https://docs.ultralytics.com/usage/cfg/#augmentation

You can change any of those directly in the yaml, or create a new yaml and then point to it, i.e. `yolo train cfg=new_cfg.yaml`

**Comment by Zengyf-CVer** [comment]
@glenn-jocher I want to set it through the Python code. Is there a cfg parameter in `model.train`?
![image](https://github.com/ultralytics/ultralytics/assets/41098760/4259f7e1-031b-4945-a74e-364a2d3d44a2)

**Comment by glenn-jocher** [comment]
@Zengyf-CVer yes, you can set the augmentation parameters through the `data` argument in `model.train(data)` function. 

This argument takes in a dictionary of configurations for the data loader, including the `train` dictionary, where you can specify the augmentation settings. For example, you can set `train: jitter: 0.3`, which will randomly resize the image by 30%. 

You can also specify other augmentation settings in the `train` dictionary such as hue, saturation, exposure, and more. 

The `data` argument can be modified within your Python code to customize the augmentation settings for your YOLOv8 training.

**Comment by Zengyf-CVer** [comment]
@glenn-jocher I don't quite understand the usage of `train dictionary` you mentioned. I mean, can I use `model.train(cfg="./new_cfg.yaml")` directly? In addition, I did not find the method of using `train: jitter: 0.3` you mentioned in your documentation `https://docs.ultralytics.com/modes/train/#arguments`.

**Comment by glenn-jocher** [comment]
Yes, you can do model.train(cfg=...), which is the same as if you had passed all the args in the cfg yaml individually, or 'yolo train cfg=...'

Any arguments you pass after cfg will override values inside it, ie epochs=10

**Comment by 33sakib33** [comment]
is this yaml correct ?
file_content = """
path: # dataset root dir
train:   # train images (relative to 'path')
val:   # val images (relative to 'path') 
test:  #test images

# Classes
names:
  0: class1
  1: class2
  2: class3
  3: class4
  4: class5
augmentations:
  hsv_v: .8
  hsv_s: .6  
  copy_paste: .03
  
If this is incorrect, can you give an example of applying augmentation to training? Thank you. 
and where do I put the hyperparameters for finetuning yolov8?

**Comment by glenn-jocher** [comment]
@Zengyf-CVer yes, the provided YAML is correct. It defines the paths for the train, val, and test images, as well as the names for each class.

To apply augmentations during training, you can specify them under the "augmentations" section in the YAML file. In this example, the "hsv_v" augmentation is set to 0.8 (adjusting the value component of the HSV color space), the "hsv_s" is set to 0.6 (adjusting the saturation component), and the "copy_paste" augmentation is set to 0.03 (introducing random copy-paste augmentation).

For hyperparameters related to fine-tuning YOLOv8, they can be set as additional arguments passed to the `model.train()` function in your Python code. Some commonly used hyperparameters include the learning rate, weight decay, batch size, and number of epochs. These hyperparameters control the training process and influence the model's performance during fine-tuning.

Please let me know if you need any further assistance or clarification.
